<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
 <html><head><title>R: Calculate Normalization Factors to Align Columns of a Count...</title>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
 <link rel="stylesheet" type="text/css" href="R.css">
 </head><body>
 
 <table width="100%" summary="page for calcNormFactors"><tr><td>calcNormFactors</td><td align="right">R Documentation</td></tr></table>
 
 <h2>Calculate Normalization Factors to Align Columns of a Count Matrix</h2>
 
 <h3>Description</h3>
 
 
 <p>Calculate normalization factors to scale the raw library sizes.
 </p>
 
 
 <h3>Usage</h3>
 
 <pre>
 calcNormFactors(object, method=c("TMM","RLE","upperquartile"), refColumn = NULL, logratioTrim = .3, sumTrim = 0.05, doWeighting=TRUE, Acutoff=-1e10, p=0.75)
 </pre>
 
 
 <h3>Arguments</h3>
 
 
 <table summary="R argblock">
 <tr valign="top"><td><code>object</code></td>
 <td>
 <p>either a <code>matrix</code> of raw (read) counts or a <code>DGEList</code> object</p>
 </td></tr>
 <tr valign="top"><td><code>method</code></td>
 <td>
 <p>method to use to calculate the scale factors</p>
 </td></tr>
 <tr valign="top"><td><code>refColumn</code></td>
 <td>
 <p>column to use as reference for <code>method="TMM"</code></p>
 </td></tr>
 <tr valign="top"><td><code>logratioTrim</code></td>
 <td>
 <p>amount of trim to use on log-ratios (&quot;M&quot; values) for <code>method="TMM"</code></p>
 </td></tr>
 <tr valign="top"><td><code>sumTrim</code></td>
 <td>
 <p>amount of trim to use on the combined absolute levels (&quot;A&quot; values) for <code>method="TMM"</code></p>
 </td></tr>
 <tr valign="top"><td><code>doWeighting</code></td>
 <td>
 <p>logical, whether to compute (asymptotic binomial precision) weights for <code>method="TMM"</code></p>
 </td></tr>
 <tr valign="top"><td><code>Acutoff</code></td>
 <td>
 <p>cutoff on &quot;A&quot; values to use before trimming for <code>method="TMM"</code></p>
 </td></tr>
 <tr valign="top"><td><code>p</code></td>
 <td>
 <p>percentile (between 0 and 1) of the counts that is aligned when <code>method="upperquartile"</code></p>
 </td></tr>
 </table>
 
 
 <h3>Details</h3>
 
 
 <p><code>method="TMM"</code> is the weighted trimmed mean of M-values (to the reference) proposed by Robinson and Oshlack (2010), where the weights are from the delta method on Binomial data.  If <code>refColumn</code> is unspecified, the library whose upper quartile is closest to the mean upper quartile is used.
 </p>
 <p><code>method="RLE"</code> is the scaling factor method proposed by Anders and Huber (2010).
 We call it &quot;relative log expression&quot;, as median library is calculated from the geometric mean of all columns and the median ratio of each sample to the median library is taken as the scale factor.
 </p>
 <p><code>method="upperquartile"</code> is the upper-quartile normalization method of Bullard et al (2010), in which the scale factors are calculated from the 75% quantile of the counts for each library,
 after removing transcripts which are zero in all libraries.
 This idea is generalized here to allow scaling by any quantile of the distributions.
 </p>
 <p>For symmetry, normalization factors are adjusted to multiply to 1.
 The effective library size is then the original library size multiplied by the scaling factor.
 </p>
 
 
 <h3>Value</h3>
 
 
 <p>If a <code>matrix</code> is given for <code>object</code>, the output is a vector with length <code>ncol(object)</code> giving the relative normalization factors.
 If a <code>DGEList</code> object is given for <code>object</code>, the output is a <code>DGEList</code> object containing the normalization factors in the <code>samples$norm.factors</code> element.
 </p>
 
 
 <h3>Author(s)</h3>
 
 <p>Mark Robinson, Gordon Smyth</p>
 
 
 <h3>References</h3>
 
 
 <p>Anders, S, Huber, W (2010).
 Differential expression analysis for sequence count data
 <EM>Genome Biology</EM> 11, R106.
 </p>
 <p>Bullard JH, Purdom E, Hansen KD, Dudoit S. (2010)
 Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments.
 <EM>BMC Bioinformatics</EM> 11, 94.
 A scaling normalization method for differential expression analysis of RNA-seq data.
 </p>
 <p>Robinson MD, Oshlack A (2010).
 <EM>Genome Biology</EM> 11, R25.
 </p>
 
 
 <h3>Examples</h3>
 
 <pre>
 y &lt;- matrix( rpois(1000, lambda=5), nrow=200 )
 calcNormFactors(y)
 </pre>
 
 
 </body></html>
