<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
 <html><head><title>R: Compute overlap encodings</title>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
 <link rel="stylesheet" type="text/css" href="R.css">
 </head><body>
 
 <table width="100%" summary="page for encodeOverlaps"><tr><td>encodeOverlaps</td><td align="right">R Documentation</td></tr></table>
 
 <h2>Compute overlap encodings</h2>
 
 <h3>Description</h3>
 
 
 <p>The <code>encodeOverlaps</code> function computes the overlap encodings
 between a <code>query</code> and a <code>subject</code>, both list-like objects
 with top-level elements typically containing multiple ranges.
 </p>
 
 
 <h3>Usage</h3>
 
 <pre>
 encodeOverlaps(query, subject, hits=NULL)
 </pre>
 
 
 <h3>Arguments</h3>
 
 
 <table summary="R argblock">
 <tr valign="top"><td><code>query, subject</code></td>
 <td>
 
 <p>List-like objects, usually of the same length, with top-level elements
 typically containing multiple ranges (e.g. RangesList objects).
 If the 2 objects don't have the same length, and if <code>hits</code> is not
 supplied, then the shortest is recycled to the length of the longest
 (the standard recycling rules apply).
 </p>
 </td></tr>
 <tr valign="top"><td><code>hits</code></td>
 <td>
 
 <p>An optional Hits object that is typically the result of a call
 to <code>findOverlaps(query, subject)</code>.
 Supplying <code>hits</code> is a convenient way to do
 <code>encodeOverlaps(query[queryHits(hits)], subject[subjectHits(hits)])</code>,
 that is,
 calling <code>encodeOverlaps(query, subject, hits)</code> will perform the
 above.
 In that case, <code>query</code> and <code>subject</code> are not expected to have
 the same length anymore.
 </p>
 </td></tr>
 </table>
 
 
 <h3>Details</h3>
 
 
 <p>See <code>?OverlapEncodings</code> for a short introduction to
 &quot;overlap encodings&quot;.
 </p>
 
 
 <h3>Value</h3>
 
 
 <p>An OverlapEncodings object with the length of <code>query</code>
 and <code>subject</code> for <code>encodeOverlaps(query, subject)</code>,
 or with the length of <code>hits</code> for
 <code>encodeOverlaps(query, subject, hits)</code>.
 </p>
 
 
 <h3>Author(s)</h3>
 
 <p>H. Pages</p>
 
 
 <h3>See Also</h3>
 
 
 <p><code>OverlapEncodings</code>,
 isCompatibleWithSplicing,
 Hits-class,
 findOverlaps,
 RangesList-class
 </p>
 
 
 <h3>Examples</h3>
 
 <pre>
 ## ---------------------------------------------------------------------
 ## A. BETWEEN 2 RangesList OBJECTS
 ## ---------------------------------------------------------------------
 ## In the context of an RNA-seq experiment, encoding the overlaps
 ## between 2 GRangesList objects, one containing the reads (the query),
 ## and one containing the transcripts (the subject), can be used for
 ## detecting hits between reads and transcripts that are "compatible"
 ## with the splicing of the transcript. Here we illustrate this with 2
 ## RangesList objects, in order to keep things simple:
 
 ## 4 aligned reads in the query:
 read1 &lt;- IRanges(c(7, 15, 22), c(9, 19, 23))  # 2 gaps
 read2 &lt;- IRanges(c(5, 15), c(9, 17))  # 1 gap
 read3 &lt;- IRanges(c(16, 22), c(19, 24))  # 1 gap
 read4 &lt;- IRanges(c(16, 23), c(19, 24))  # 1 gap
 query &lt;- IRangesList(read1, read2, read3, read4)
 
 ## 1 transcript in the subject:
 tx &lt;- IRanges(c(1, 4, 15, 22, 38), c(2, 9, 19, 25, 47))  # 5 exons
 subject &lt;- IRangesList(tx)
 
 ## Encode the overlaps:
 ovenc &lt;- encodeOverlaps(query, subject)
 ovenc
 encoding(ovenc)
 
 ## Reads that are "compatible" with the transcript can be detected with
 ## a regular expression (the regular expression below assumes that
 ## reads have at most 2 gaps):
 regex0 &lt;- "(:[fgij]:|:[jg].:.[gf]:|:[jg]..:.g.:..[gf]:)"
 grepl(regex0, encoding(ovenc))  # read4 is NOT "compatible"
 
 ## This was for illustration purpose only. In practise you don't need
 ## (and should not) use this regular expression, but use instead the
 ## isCompatibleWithSplicing() utility function defined in the
 ## GenomicRanges package. See '?isCompatibleWithSplicing' in the
 ## GenomicRanges package for more information.
 
 ## ---------------------------------------------------------------------
 ## B. BETWEEN 2 GRangesList OBJECTS
 ## ---------------------------------------------------------------------
 ## With real RNA-seq data, the reads and transcripts will typically be
 ## stored in GRangesList objects. See '?isCompatibleWithSplicing' in the
 ## GenomicRanges package for more information.
 </pre>
 
 
 </body></html>
