<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
 <html><head><title>R: Genewise Negative Binomial Generalized Linear Mdels</title>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
 <link rel="stylesheet" type="text/css" href="R.css">
 </head><body>
 
 <table width="100%" summary="page for glmFit"><tr><td>glmFit</td><td align="right">R Documentation</td></tr></table>
 
 <h2>Genewise Negative Binomial Generalized Linear Mdels</h2>
 
 <h3>Description</h3>
 
 <p>Fit a negative binomial generalized log-linear model to the read counts for each gene or transcript.
 Conduct statistical tests based on the fitted models.</p>
 
 
 <h3>Usage</h3>
 
 <pre>
 ## S3 method for class 'DGEList'
 glmFit(y, design=NULL, dispersion=NULL, offset=NULL, weights=NULL, lib.size=NULL, prior.count.total=0.5, start=NULL, method="auto", ...)
 glmLRT(y, glmfit, coef=ncol(glmfit$design), contrast=NULL)
 glmQLFTest(y, glmfit, coef=ncol(glmfit$design), contrast=NULL, abundance.trend=TRUE)
 </pre>
 
 
 <h3>Arguments</h3>
 
 
 <table summary="R argblock">
 <tr valign="top"><td><code>y</code></td>
 <td>
 <p>an object that contains the raw counts for each library (the measure of expression level); alternatively, a matrix of counts, or a <code>DGEList</code> object with (at least) elements <code>counts</code> (table of unadjusted counts) and <code>samples</code> (data frame containing information about experimental group, library size and normalization factor for the library size)</p>
 </td></tr>
 <tr valign="top"><td><code>design</code></td>
 <td>
 <p>numeric matrix giving the design matrix for the tagwise linear models.
 Must be of full column rank.
 Defaults to a single column of ones, equivalent to treating the columns as replicate libraries.</p>
 </td></tr>
 <tr valign="top"><td><code>dispersion</code></td>
 <td>
 <p>numeric scalar or vector providing the value for the dispersion parameter that is used in fitting the GLM for each transcript. Can be a common value for all tags, or a vector of values can provide a unique dispersion value for each tag. If <code>NULL</code> (default) then dispersion will be detected and extracted from <code>y</code>, if possible, with order of precedence: tagwise dispersion, trended dispersions, common dispersion.</p>
 </td></tr>
 <tr valign="top"><td><code>offset</code></td>
 <td>
 <p>numeric scalar, vector or matrix giving the offset that is to be included in the NB GLM for the transcripts. Only one of <code>offset</code> and <code>lib.size</code> should be supplied&mdash;if both are supplied then <code>offset</code> will be used and <code>lib.size</code> will be ignored. If a scalar, then this value will be used as an offset for all transcripts and libraries. If a vector, it should be have length equal to the number of libraries, and the same vector of offsets will be used for each transcript. If a matrix, then each library for each transcript can have a unique offset, if desired.
 Defaults to the log normalized library sizes.</p>
 </td></tr>
 <tr valign="top"><td><code>weights</code></td>
 <td>
 <p>optional numeric matrix giving prior weights for the observations (for each library and transcript) to be used in the GLM calculations.  Not supported by methods <code>"linesearch"</code> or <code>"levenberg"</code>.</p>
 </td></tr>
 <tr valign="top"><td><code>lib.size</code></td>
 <td>
 <p>optional numeric vector providing the (effective) library size for each library (must have length equal to the number of columns, or libraries, in the matrix of counts). If <code>NULL</code>, then a default is used. If <code>y</code> is a <code>DGEList</code> object then the default for <code>lib.size</code> is the product of the library sizes and the normalization factors (in the <code>samples</code> slot of the object). If <code>y</code> is a simple matrix of counts, then the default for <code>lib.size</code> is the vector of column sums of <code>y</code>.</p>
 </td></tr>
 <tr valign="top"><td><code>prior.count.total</code></td>
 <td>
 <p>the total number of prior counts to be augmented to the data to shrink the estimated log-fold-changes.</p>
 </td></tr>
 <tr valign="top"><td><code>start</code></td>
 <td>
 <p>optional numeric matrix of initial estimates for the fitted coefficients.</p>
 </td></tr>
 <tr valign="top"><td><code>method</code></td>
 <td>
 <p>which fitting algorithm to use.  Possible values are <code>"auto"</code>, <code>"linesearch"</code>, <code>"levenberg"</code> or <code>"simple"</code>.</p>
 </td></tr>
 <tr valign="top"><td><code>...</code></td>
 <td>
 <p>other arguments are passed to lower-level functions, for example to <code>mglmLS</code>.</p>
 </td></tr>
 <tr valign="top"><td><code>glmfit</code></td>
 <td>
 <p>a <code>DGEGLM</code> object, the output from <code>glmFit</code>.</p>
 </td></tr>
 <tr valign="top"><td><code>coef</code></td>
 <td>
 <p>scalar or vector indicating the column(s) of <code>design</code> that are to be dropped when creating the null model for the Likelihood Ratio (LR) Test. Can be numeric or character. If character, the string(s) provided to <code>coef</code> must match a column of the design matrix in the <code>glmfit</code> object passed to <code>glmLRT</code>. The <code>glmLRT</code> fits the null model and then conducts an LR test of the model fit provided in <code>glmfit</code> against the null model defined by the choice of <code>coef</code>. By default, the last column of the design matrix is dropped to form the design matrix for the null model.</p>
 </td></tr>
 <tr valign="top"><td><code>contrast</code></td>
 <td>
 <p>contrast vector for which the test is required, of length equal to the number of columns of <code>design</code>. If specified, then takes precedence over <code>coef</code>.</p>
 </td></tr>
 <tr valign="top"><td><code>abundance.trend</code></td>
 <td>
 <p>logical, whether to allow an abundance-dependent trend when estimating the prior values for the quasi-likelihood multiplicative dispersion parameter.</p>
 </td></tr>
 </table>
 
 
 <h3>Details</h3>
 
 
 <p><code>glmFit</code> and <code>glmLRT</code> implement methods developed by McCarthy et al (2012).
 </p>
 <p><code>glmFit</code> fits genewise negative binomial generalized linear models (glms), all with the same design matrix but possibly different dispersions, offsets and weights.
 When the design matrix defines a one-way layout, or can be re-parametrized to a one-way layout, the glms are fitting very quickly using <code>mglmOneGroup</code>.
 Otherwise the default fitting method, implemented in <code>mglmLS</code>, is a parallelized line search algorithm described by McCarthy et al (2012).
 Other possible fitting methods are <code>mglmLevenberg</code> and <code>mglmSimple</code>.
 </p>
 <p><code>glmLRT</code> conducts likelihood ratio tests for one or more coefficients in the linear model.
 </p>
 <p><code>glmQLFTest</code> conducts quasi-likelihood F-tests for one or more coefficients in the linear model.
 This function calls the limma function <code>squeezeVar</code> to conduct empirical Bayes smoothing of the genewise multiplicative dispersions.
 </p>
 
 
 <h3>Value</h3>
 
 
 <p>object of class <code>DGEGLM</code> with the following components:
 </p>
 <table summary="R valueblock">
 <tr valign="top"><td><code>coefficients</code></td>
 <td>
 <p>numeric matrix of estimated coefficients from the tagwise linear models, on the natural log scale, of size <code>nrow(y)</code> by <code>ncol(design)</code>.</p>
 </td></tr>
 <tr valign="top"><td><code>df.residual</code></td>
 <td>
 <p>numeric vector of giving residual degrees of freedom for each tag.</p>
 </td></tr>
 <tr valign="top"><td><code>deviance</code></td>
 <td>
 <p>numeric vector giving the NB deviance from the model fit for each tag.</p>
 </td></tr>
 <tr valign="top"><td><code>design</code></td>
 <td>
 <p>design matrix used in the NB model fit for each tag.</p>
 </td></tr>
 <tr valign="top"><td><code>offset</code></td>
 <td>
 <p>numeric matrix of linear model offsets.</p>
 </td></tr>
 <tr valign="top"><td><code>samples</code></td>
 <td>
 <p>data frame of information about the samples (libraries) in the experiment; taken from the object <code>y</code>.</p>
 </td></tr>
 <tr valign="top"><td><code>genes</code></td>
 <td>
 <p>data frame of tag annotatoi information for each tag; taken from the object <code>y</code>.</p>
 </td></tr>
 <tr valign="top"><td><code>dispersion</code></td>
 <td>
 <p>scalar or vector giving the the value of the dispersion parameter used in each tag's NB model fit.</p>
 </td></tr>
 <tr valign="top"><td><code>lib.size</code></td>
 <td>
 <p>vector of library sizes used in the model fit.</p>
 </td></tr>
 <tr valign="top"><td><code>weights</code></td>
 <td>
 <p>matrix of final weights used in the NB model fits for each tag.</p>
 </td></tr>
 <tr valign="top"><td><code>fitted.values</code></td>
 <td>
 <p>matrix of fitted values from the NB model for each tag.</p>
 </td></tr>
 <tr valign="top"><td><code>abundance</code></td>
 <td>
 <p>vector of gene/tag abundances (expression level), on the log2 scale, computed from the mean count for each gene/tag after scaling count by normalized library size.</p>
 </td></tr>
 </table>
 <p><code>glmLRT</code> produces an object of class <code>DGELRT</code> with the following components:
 </p>
 <table summary="R valueblock">
 <tr valign="top"><td><code>table</code></td>
 <td>
 <p>data frame (table) containing the abundance of each tag (log-concentration, <code>logConc</code>), the log2-fold change of expression between conditions/contrasts being tested (<code>logFC</code>), the likelihood ratio statistic (<code>LR.statistic</code>) and the p-value from the LR test (<code>p.value</code>), for each tag in the dataset.</p>
 </td></tr>
 <tr valign="top"><td><code>coefficients</code></td>
 <td>
 <p>matrix of coefficients for the full model defined by the <code>design</code> matrix (i.e. for the full model).</p>
 </td></tr>
 <tr valign="top"><td><code>dispersion.used</code></td>
 <td>
 <p>scalar or vector of the dispersion value(s) used in the GLM fits and LR test.</p>
 </td></tr>
 </table>
 <p>The <code>DGELRT</code> object also contains all the elements of <code>y</code> except for the table of counts (raw data) and the table of pseudo-counts (if applicable).
 </p>
 <p><code>glmQLFtest</code> produces an object of class <code>DGELRT</code> with the following components:
 </p>
 <table summary="R valueblock">
 <tr valign="top"><td><code>table</code></td>
 <td>
 <p>data frame (table) containing the abundance of each tag (log-concentration, <code>logConc</code>), the log2-fold change of expression between conditions/contrasts being tested (<code>logFC</code>), the quasi-likelihood F statistic (<code>F</code>) and the p-value from the QL F test (<code>p.value</code>), for each tag in the dataset.</p>
 </td></tr>
 <tr valign="top"><td><code>coefficients</code></td>
 <td>
 <p>matrix of coefficients for the full model defined by the <code>design</code> matrix (i.e. for the full model).</p>
 </td></tr>
 <tr valign="top"><td><code>dispersion.used</code></td>
 <td>
 <p>scalar or vector of the dispersion value(s) used in the GLM fits and LR test.</p>
 </td></tr>
 </table>
 <p>The <code>DGELRT</code> object also contains all the elements of <code>y</code> except for the table of counts (raw data) and the table of pseudo-counts (if applicable).
 </p>
 
 
 <h3>Author(s)</h3>
 
 <p>Davis McCarthy and Gordon Smyth</p>
 
 
 <h3>References</h3>
 
 
 <p>McCarthy, DJ, Chen, Y, Smyth, GK (2012). Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation.
 <EM>Nucleic Acids Research</EM>.
 <a href="http://nar.oxfordjournals.org/content/early/2012/02/06/nar.gks042">http://nar.oxfordjournals.org/content/early/2012/02/06/nar.gks042</a>
 (Published online 28 January 2012)
 </p>
 
 
 <h3>See Also</h3>
 
 
 <p>Low-level computations are done by <code>mglmOneGroup</code>, <code>mglmLS</code>, <code>mglmLevenberg</code> or <code>mglmSimple</code>.
 </p>
 <p>See <code>topTags</code> for displaying results from <code>glmLRT</code>.
 </p>
 
 
 <h3>Examples</h3>
 
 <pre>
 nlibs &lt;- 3
 ntags &lt;- 100
 dispersion.true &lt;- 0.1
 
 # Make first transcript respond to covariate x
 x &lt;- 0:2
 design &lt;- model.matrix(~x)
 beta.true &lt;- cbind(Beta1=2,Beta2=c(2,rep(0,ntags-1)))
 mu.true &lt;- 2^(beta.true %*% t(design))
 
 # Generate count data
 y &lt;- rnbinom(ntags*nlibs,mu=mu.true,size=1/dispersion.true)
 y &lt;- matrix(y,ntags,nlibs)
 colnames(y) &lt;- c("x0","x1","x2")
 rownames(y) &lt;- paste("Gene",1:ntags,sep="")
 d &lt;- DGEList(y)
 
 # Normalize
 d &lt;- calcNormFactors(d)
 
 # Fit the NB GLMs
 fit &lt;- glmFit(d, design, dispersion=dispersion.true)
 
 # Likelihood ratio tests for trend
 results &lt;- glmLRT(d, fit, coef=2)
 topTags(results)
 
 # Estimate the dispersion (may be unreliable with so few tags)
 d &lt;- estimateGLMCommonDisp(d, design, verbose=TRUE)
 </pre>
 
 
 </body></html>
