<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
 <html><head><title>R: Plot of regularized linear discriminant functions for...</title>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
 <link rel="stylesheet" type="text/css" href="R.css">
 </head><body>
 
 <table width="100%" summary="page for plotRLDF"><tr><td>plotRLDF</td><td align="right">R Documentation</td></tr></table>
 
 <h2>Plot of regularized linear discriminant functions for microarray data</h2>
 
 <h3>Description</h3>
 
 
 <p>Plot of regularized linear discriminant functions for microarray data.
 </p>
 
 
 <h3>Usage</h3>
 
 <pre>
 plotRLDF(y,design=NULL,z=NULL,labels.y=NULL,labels.z=NULL,col.y=1,col.z=1,
 df.prior=5,show.dimensions=c(1,2),main=NULL,nprobes=500,...)</pre>
 
 
 <h3>Arguments</h3>
 
 
 <table summary="R argblock">
 <tr valign="top"><td><code>y</code></td>
 <td>
 <p>any data object which can be coerced to a matrix, such as <code>ExpressionSet</code> or <code>EList</code>. The training dataset.</p>
 </td></tr>
 <tr valign="top"><td><code>z</code></td>
 <td>
 <p>any data object which can be coerced to a matrix, such as <code>ExpressionSet</code> or <code>EList</code>. The dataset to be classified.</p>
 </td></tr>
 <tr valign="top"><td><code>design</code></td>
 <td>
 <p>the design matrix ofthe microarray experiment for <code>y</code>, with rows corresponding to arrays and columns to coefficients to be estimated. Defaults to the unit vector meaning that the arrays are treated as replicates.</p>
 </td></tr>
 <tr valign="top"><td><code>labels.y</code></td>
 <td>
 <p>character vector of sample names or labels in <code>y</code>. Default is integers starting from 1.</p>
 </td></tr>
 <tr valign="top"><td><code>labels.z</code></td>
 <td>
 <p>character vector of sample names or labels in <code>z</code>. Default is <code>letters</code>.</p>
 </td></tr>
 <tr valign="top"><td><code>col.y</code></td>
 <td>
 <p>numeric or character vector of colors for the plotting characters of <code>y</code>. Default is black.</p>
 </td></tr>
 <tr valign="top"><td><code>col.z</code></td>
 <td>
 <p>numeric or character vector of colors for the plotting characters of <code>z</code>. Default is black.</p>
 </td></tr>
 <tr valign="top"><td><code>df.prior</code></td>
 <td>
 <p>prior degrees of freedom for residual variances. Used in gene selection.</p>
 </td></tr>
 <tr valign="top"><td><code>show.dimensions</code></td>
 <td>
 <p>which two dimensions should be plotted, numeric vector of length two.</p>
 </td></tr>
 <tr valign="top"><td><code>main</code></td>
 <td>
 <p>title of the plot.</p>
 </td></tr>
 <tr valign="top"><td><code>nprobes</code></td>
 <td>
 <p>number of probes to be used for the calculations. Selected by moderated F tests.</p>
 </td></tr>
 <tr valign="top"><td><code>...</code></td>
 <td>
 <p>any other arguments are passed to <code>plot</code>.</p>
 </td></tr>
 </table>
 
 
 <h3>Details</h3>
 
 
 <p>This function is a variation on the plot of usual linear discriminant fuction, in that the within-group covariance matrix is regularized to ensure that it is invertible, with eigenvalues bounded away from zero.
 A diagonal regulation using <code>df.prior</code> and the median within-group variance is used.
 </p>
 <p>The calculations are based on a filtered list of probes.
 The <code>nprobes</code> probes with largest moderated F statistics are used to discriminate.
 </p>
 <p>See <code>text</code> for possible values for <code>col</code> and <code>cex</code>.
 </p>
 
 
 <h3>Value</h3>
 
 <p>A list containing metagene information is (invisibly) returned.
 A plot is created on the current graphics device.</p>
 
 
 <h3>Author(s)</h3>
 
 <p>Di Wu and Gordon Smyth</p>
 
 
 <h3>See Also</h3>
 
 
 <p><code>lda</code> in package <code>MASS</code>
 </p>
 
 
 <h3>Examples</h3>
 
 <pre>
 
 # Simulate gene expression data for 1000 probes and 6 microarrays.
 # Samples are in two groups
 # First 50 probes are differentially expressed in second group
 sd &lt;- 0.3*sqrt(4/rchisq(1000,df=4))
 y &lt;- matrix(rnorm(1000*6,sd=sd),1000,6)
 rownames(y) &lt;- paste("Gene",1:1000)
 y[1:50,4:6] &lt;- y[1:50,4:6] + 2
 
 z &lt;- matrix(rnorm(1000*6,sd=sd),1000,6)
 rownames(z) &lt;- paste("Gene",1:1000)
 z[1:50,4:6] &lt;- z[1:50,4:6] + 1.8
 z[1:50,1:3] &lt;- z[1:50,1:3] - 0.2
 
 design &lt;- cbind(Grp1=1,Grp2vs1=c(0,0,0,1,1,1))
 options(digit=3)
 
 plotRLDF(y,z, design=design)
 </pre>
 
 
 </body></html>
